### PR TITLE
Add pixlstash-cli plugins test: check a captioning plugin without a restart

### PR DESCRIPTION
**Pull request by lindkvis**
Closes Pikselkroken/pixlstash-plugins#4, as a CLI verb rather than the standalone
`scripts/check_plugin.py` that issue asked for.

## Why

Discovery runs once, at start-up. So the loop for finding a typo in a captioning plugin is
edit → restart the server → wait for the boot → read the error row under Settings ›
Auto-tagging, and that is every contributor's first experience of writing one.

```
pixlstash-cli plugins test ./my_captioner.py [--image sample.jpg]
```

A verb rather than a script because the CLI already owns this domain — `plugins
install|list|remove` are here, the destination logic is here, and a script in `scripts/`
would have been a fourth thing to find and the only one not in `--help`.

## What it checks

- **It imports the plugin the way the server does.** Not a lookalike loader:
  `TaggerPluginManager.load_plugin_from_path` was extracted out of the body of
  `_load_user_plugins`, which now calls it. Same module namespacing, same
  `submodule_search_locations` for a package (so `from . import helper` resolves), same
  `sys.modules` registration before `exec_module`, same containment of a failing import.
  A second implementation that got the package case wrong would fail plugins that work in
  the server, which is worse than no check at all.
- **The decisions *around* the loader are not shared, and are restated.** Discovery filters
  the directory listing above that method and refuses duplicate names below it, and the
  checker's manager scans no directory, so neither fires by itself: a `_`- or
  `.`-prefixed entry, and a name an installed plugin already holds, are both checked
  explicitly. Without that a plugin passes here and silently never loads there — the exact
  failure the command exists to remove.
- **The parameter schema, which nothing else checks.** The registry exercises
  `plugin_schema()` at registration so a raising plugin cannot take out the boot, but a
  schema that *renders* wrong raises nowhere. An unknown `type` falls through
  `TaggerParametersUI.vue`'s `v-else` to a text box; a `select` with no `options`/`enum`
  fails that branch's `Array.isArray` guard and becomes a text box too; `options: []`
  satisfies the guard and gives a real dropdown with nothing in it. A missing `default` or
  `name` is worse — `default_params()` and `fill_defaults()` read both unguarded on every
  library open. A first-party name collision is checked too, since the built-in wins it.
- **`--image` runs it as the workflows do**: `setup(device)`, `init(params)`, then
  `generate_descriptions`/`tag_images` with the schema's defaults, checking the result is a
  dict keyed by the paths it was given (the workflow looks results up by path and drops
  anything else).

## What it is not

**A development aid, not a security scanner**, said in the `--help`, in the output *before*
the plugin runs, in the README, in the plugin guide, in the template and in
`backend_architecture.md` §8.1. It cannot tell you whether a plugin is safe, because
finding out whether it loads means running it — unsandboxed, with your permissions,
exactly as the server would. A report a user reads as a safety verdict would be worse than
no report.

Observing what a plugin *reaches for* (an audit hook over `socket.connect` /
`subprocess.Popen` / writes) is disclosure rather than containment and is worth doing, but
separately and with a security review; filed as its own issue.

`--image` stops when `needs_download()` is True rather than starting a multi-gigabyte fetch
from a check command. That is deliberately worded as a courtesy and not a guarantee
everywhere it appears: `needs_download()` is the plugin's own answer about its own files,
and a plugin that downloads inside `init()` — where `from_pretrained_local_first` does it,
and so where an author copying the shipped captioners will do it — is already past that
gate.

Image filters are not checked: different base class, different parameter schema (`string` +
`enum`, no `select`). Pointed at one, the command says so rather than reporting "No
TaggerPlugin subclass found", which is true and useless.

## Tests

Driven off the shipped `tagger_plugins/plugin_template.py`, so the fixtures cannot drift
from what people actually write: the pass case guards the starter we hand contributors, and
each failure case is one mistake spliced into a copy, anchored on a string asserted to
occur exactly once so a mutation that hit a docstring instead of code cannot read as
coverage.

Two guardrails worth calling out:

- `test_schema_types_match_the_component_that_renders_them` parses
  `TaggerParametersUI.vue` and pins `SCHEMA_TYPES` to its `v-else-if` chain. A hand-copied
  list mirroring a Vue component is exactly the drift `_SUBDIRS`-pinned-to-the-registries
  exists to forbid, and this is the one piece of genuinely new checking here.
- The `--image` test splices sentinels into its copy rather than running the template as
  shipped. As shipped it *cannot* fail: the template seeds `self._device = "cpu"` in
  `__init__` and falls back to `max_tokens or 128`, so a working run's caption is
  character-for-character what you get with `setup()`, `init()` and the merged defaults all
  skipped. That assertion was dead, and was found dead by deleting each of the three from
  `plugin_check` in turn. With the sentinels in, all three deletions fail and the unmutated
  file passes.

No new test file, so no CI shard list change: the tests live in `tests/test_plugin_install.py`,
which is already gated and already the home of the `plugins` verbs.

## Review findings answered rather than actioned

An adversarial pass over this diff raised these; I disagree with them and am recording why
rather than dropping them silently.

- **"The first-party collision check is a second implementation, contradicting the
  no-reimplementation claim."** It calls `plugin_install.builtin_names`, the existing shared
  helper `plugins install` uses for the same question. Reusing it is the opposite of
  reimplementing; the docstring's claim is about the *loader*, and I have reworded it so
  that reads unambiguously.
- **"The registry logs a traceback to stderr ahead of the friendly message."** It does, and
  for this command that traceback points at the failing line *in the plugin being written*,
  which is the single most useful thing on screen for the person debugging it. I would not
  suppress it here.
- **"Two of the mutation cases exercise pre-existing registry behaviour, not the new
  module."** True. `plugins test` is a command, and its contract includes reporting an
  import failure and a file with no plugin class in it; those cases are testing the command
  rather than the module, deliberately.
- **"`plugins test` exits 0 on `plugin_template.py`, which `plugins install` refuses."**
  Install refuses that *filename*; the template's own step 2 is "rename it". Checking the
  template before renaming it is the right answer, not a contradiction.
- **"309 + 161 lines for ~30 lines of new checking."** The line count is mostly the loader
  seam, the wrong-kind and eligibility branches and the docstrings that record why each
  exists. I would rather have those than rediscover the `_`-prefix rule.

### CODE REVIEW

File: pixlstash/plugin_check.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+
+    Returns:
+        ``(what came back, problems)``. The result is ``None`` when the plugin
+        never got as far as returning anything.
+    """
+    image_path = str(Path(image).expanduser().resolve())
+    parameters = plugin.default_params()
+
+    try:
+        if plugin.needs_download(parameters):
```
[the remark this answers is not in the document]
Comment (lindkvis) [reply pixlstash/plugin_check.py:388]: Right, and fixed in 0aea23d7. Which method to call is now decided at the top of `_run_over_image`, above `needs_download()` and `init()`: a plugin with neither flag returns immediately, so nothing is downloaded and no model is loaded for a call that was never going to happen. The workflows would never reach such a plugin either, so this now matches them.

`test_image_loads_no_model_for_a_plugin_nothing_will_ever_call` pins it — the template copy it uses has `init()` raise, so if init is reached at all the run reports `init() raised` and exits 1. I checked it has teeth by moving the capability block back below init: the test fails, and passes again with it restored.

File: docs/writing-tagger-plugins.md [hunk 2]
```diff
@@ -40,6 +40,38 @@ Entries whose name starts with `.` or `_` are skipped, as is any other file type
...
+the settings screen can actually render (§3), which the server does not check. `--image`
+also runs the plugin over one picture with your defaults and prints what came back.
+
+It asks your `needs_download()` first and stops if you answer yes, so a check does not
+start a multi-gigabyte fetch nobody asked for — but that is a courtesy and not a
+guarantee. If your `init()` downloads (which is what `from_pretrained_local_first` does,
+and therefore what the shipped captioners do), it downloads here too. Fetch models from
+**Settings → Auto-tagging** first if you would rather it did not.
+
+It exits `1` on anything it found.
```
[the remark this answers is not in the document]
Comment (lindkvis) [reply docs/writing-tagger-plugins.md]: Correct — that sentence predates the problem/warning split and was left describing the older behaviour. Fixed in 0aea23d7, in the guide and in `--help`, which had the same stale claim ("exits 1 if anything failed"):

> A **problem** — it did not load, a parameter has no `name` or `default`, a `type` that will not render, a name a built-in or an installed plugin already holds — exits `1`. A **warning** — a parameter with no `label`, or neither capability flag set — is printed and exits `0`, because those plugins do work.

`test_a_cosmetic_fault_is_reported_without_failing_the_command` already asserted the `0`, so the code was right and only the documentation was wrong.

File: pixlstash/plugin_check.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+
+def _device() -> str:
+    """Return the device the server would hand a plugin's ``setup()``."""
+    # Local import: torch is seconds of start-up, and every other verb in this
+    # CLI — including `plugins test` without --image — runs without it.
+    try:
+        import torch
+
+        return "cuda" if torch.cuda.is_available() else "cpu"
+    except ImportError:
```
[the remark this answers is not in the document]
Comment (lindkvis) [reply pixlstash/plugin_check.py]: Agreed, and fixed in 0aea23d7. `_device()` now catches `Exception`, not just `ImportError` — an installed-but-unloadable torch raises `OSError` (a missing CUDA shared library being the usual way) and a partial install can raise almost anything, and all of them mean the same thing here.

It is not swallowed: the exception is logged with its type and message, so the cause survives for whoever reads the log, and the fallback is deliberate rather than a guess about the machine. A plugin that needs a device needs torch too, and its own `init()` is moments away and will report the failure in terms of its own dependency — which is a better error than a traceback out of the checker about a library the plugin author may not import directly.

`test_a_torch_that_will_not_answer_does_not_take_the_command_down` covers it; narrowing the catch back to `ImportError` fails it.
